CMakeLists.txt

```diff
@@ -331,15 +334,42 @@ if(CELERITAS_USE_OpenMP)
 endif()
 
 if(CELERITAS_USE_Perfetto)
-  find_package(Perfetto REQUIRED)
-endif()
-
-if( (CELERITAS_USE_CUDA OR CELERITAS_USE_HIP) AND CELERITAS_USE_Perfetto)
-  celeritas_error_incompatible_option(
-          "Perfetto is not supported with CUDA/HIP build"
-          CELERITAS_USE_Perfetto
-          OFF
-  )
+  if(CELERITAS_USE_CUDA OR CELERITAS_USE_HIP)
+    celeritas_error_incompatible_option(
+            "Perfetto is not supported with CUDA/HIP build"
+            CELERITAS_USE_Perfetto
+            OFF
+    )
+  endif()
+  if(PERFETTO_ROOT)
+    if(CELERITAS_PERFETTO_DOWNLOAD)
+      message(WARNING "Both CELERITAS_PERFETTO_DOWNLOAD and PERFETTO_ROOT are set. "
+              "Getting dependency from PERFETTO_ROOT.")
+    endif()
+    set(Perfetto_SDK ${PERFETTO_ROOT}/sdk)
+    if(NOT IS_DIRECTORY ${Perfetto_SDK})
+      message(FATAL_ERROR "PERFETTO_ROOT must point to the top-level perfetto directory and the sdk subdir must exist. "
+              "Make sure that you downloaded an amalgamated release")
+    endif()
+  elseif(CELERITAS_PERFETTO_DOWNLOAD)
+    include(FetchContent)
+    FetchContent_Declare(
+            perfetto
+            GIT_REPOSITORY https://android.googlesource.com/platform/external/perfetto
+            GIT_TAG 7add4fded398ef47414a556b7b81d212a4482a66 # v45.0
+    )
+    FetchContent_MakeAvailable(perfetto)
```

Comment:
can we require a newer CMake version (3.14) or is 3.11 still important to support?